internal/orchestrion/context.go

```diff
@@ -0,0 +1,70 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024 Datadog, Inc.
+
+package orchestrion
```

Comment:
IMO the fact that this is an internal package makes it effectively an implementation detail but I agree that this can be confusing hence why I wanted to reduce the usages of this package to the bare minimum (eg. dyngo and the tracer itself). Calling it `gls` alone does not really make any sense because we can't access it without orchestrion. I can concede something like `internal/orchestrion/gls` but IMO we can do this change when we have more stuff to add to do more bindings between orchestrion and dd-trace-go (and I believe we will)